hazelcast/src/main/java/com/hazelcast/cache/impl/HazelcastServerCacheManager.java

```diff
@@ -147,24 +155,32 @@ private void enableStatisticManagementOnOtherNodes(String cacheName, boolean sta
     }
 
     @Override
-    protected <K, V> CacheConfig<K, V> createCacheConfig(String cacheName,
-                                                         CacheConfig<K, V> config,
-                                                         boolean createAlsoOnOthers,
-                                                         boolean syncCreate) {
-        CacheConfig<K, V> currentCacheConfig = cacheService.getCacheConfig(cacheName);
-        OperationService operationService = nodeEngine.getOperationService();
-        // Create cache config on all nodes.
-        CacheCreateConfigOperation op =
-                new CacheCreateConfigOperation(config, createAlsoOnOthers);
-        // Run "CacheCreateConfigOperation" on this node. Its itself handles interaction with other nodes.
-        // This operation doesn't block operation thread even "syncCreate" is specified.
-        // In that case, scheduled thread is used, not operation thread.
-        InternalCompletableFuture future =
-                operationService.invokeOnTarget(CacheService.SERVICE_NAME, op, nodeEngine.getThisAddress());
-        if (syncCreate) {
-            return (CacheConfig<K, V>) future.join();
+    protected <K, V> void createCacheConfig(String cacheName,
+                                            CacheConfig<K, V> config,
+                                            boolean createAlsoOnOthers,
+                                            boolean syncCreate) {
+        Version clusterVersion = nodeEngine.getClusterService().getClusterVersion();
```

Comment:
The `syncCreate` parameter here dictates whether we should wait for the operation to complete or not before returning, otherwise it would be a good deduplication opportunity.
edit: did a quick check, seems that `syncCreate` argument is effectively always `true`. I will double check and if confirmed, replace with the call to `cacheService`.